pumpkin/src/client/container.rs

```diff
@@ -231,17 +227,17 @@ impl Player {
                             None => Some(slot_number),
                         }
                     };
-
-                    let slots = if slot > 35 {
-                        slots.skip(9).find_map(find_condition)
-                    } else {
-                        slots.skip(36).rev().find_map(find_condition)
-                    };
-                    if let Some(slot) = slots {
-                        let mut item_slot = container.all_slots()[slot].map(|i| i.to_owned());
-                        container.handle_item_change(&mut item_slot, slot, MouseClick::Left)?;
-                        *container.all_slots()[slot] = item_slot;
-                    }
+                    let all_slots = container.iter_slots_mut().enumerate();
+                    // let slots = if slot > 35 {
```

Comment:
this code works, but only for the case of being inside of Inventory. Implementing this should be easy now if we use `Container::size`